### PR TITLE
Add a workaround for the enterprise plugin

### PR DESCRIPTION
**Pull request by mlopatkin**
The plugin accesses extensions at execution time, and this now breaks the build with CC enabled. The plugin still loses the configuration information in this case, though, but the fix is already in the works.

### CODE REVIEW

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ProblemFactory.kt [hunk 1]
```diff
@@ -47,6 +48,11 @@ interface ProblemFactory {
          */
         fun exception(message: String): Builder
 
+        /**
+         * Uses a provided exception for this problem.
+         */
+        fun exception(exception: InvalidUserCodeException): Builder
```
Comment (bamboo) [inline subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ProblemFactory.kt:54]: 🤔 How is this related to the workaround?

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/Workarounds.kt [hunk 2]
```diff
@@ -53,13 +53,19 @@ object Workarounds {
 
     fun canAccessConventions(from: String, area: String) =
         withWorkaroundsFor(area) {
-            from.startsWith("com.android.build.gradle.tasks.factory.AndroidUnitTest")
+            from.startsWith("com.android.build.gradle.tasks.factory.AndroidUnitTest") || callStackHasElement {
```
Comment (bamboo) [inline subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/Workarounds.kt:56]: Nice 👍

**Review by bamboo — APPROVED** [review]

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ProblemFactory.kt (hunk 1 above)
Comment (mlopatkin) [reply subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ProblemFactory.kt]: Ouch, this is a leftover from my first attempt. Removed, thank you for the review

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Comment by mlopatkin** [comment]
@bot-gradle cancel

**Comment by mlopatkin** [comment]
@bot-gradle cancel

**Comment by mlopatkin** [comment]
@bot-gradle test and merge
